### ISSUE TITLE
Wrong directory for lamda layers in sam local start-api.

### ISSUE THREAD

**Issue by santiperone** [aws/aws-sam-cli#3787]
### Description:
In sam version 1.36.0, with nodejs 14 lambda runtime, when locally invoking a lambda or an api, layers are not built into '/opt/nodejs' directory, as they are after running sam build.

### Steps to reproduce:
import a layer from '/opt/nodejs/layerName' in your lambda function. sam local start-api will throw module not found when invoking that function. 

### Expected result:
When building the sam image from a non-built project, SAM should place the layers in /opt/nodejs. This would avoid having to build the application every time.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: MacOs Monterrey
2. `sam --version`: 1.36.0
3. AWS region: n/a

**Comment by hawflau** [comment]
Hey thanks for reporting the issue. Sorry I don't fully understand what "import a layer from '/opt/nodejs/layerName' in your lambda function" means. Can you please provide a more concrete example (e.g. template and Lambda code) for us to reproduce the issue?

**Comment by santiperone** [comment]
Sure, I created a small example to show you what I mean:
This is my template.yaml lambda and layer declaration:
`  UtilsLayer:
    Type: AWS::Serverless::LayerVersion
    Properties:
      LayerName: UtilsLayer
      Description: "Util Libs"
      ContentUri: layers/utils
      LicenseInfo: Available under the MIT-0 license.
      CompatibleRuntimes:
        - nodejs14.x
    Metadata:
      BuildMethod: nodejs14.x
  HelloWorldFunction:
    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
    Properties:
      CodeUri: functions/hello-world/
      Handler: app.lambdaHandler
      Runtime: nodejs14.x
      Architectures:
        - x86_64
      Layers:
        - Ref: UtilsLayer
      Events:
        HelloWorld:
          Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
          Properties:
            Path: /hello
            Method: get`

And my project directory looks like this: 
`
--root
----template.yaml
----functions
------hello-world
--------app.js
----layers
------utils
--------utils.js
`

The thing is, when I run `sam local start-api -t template.yaml`, my layer is built into the docker image on /opt directory, so in this case I wouldn't be able to import the layer from '/opt/nodejs/utils' in my hello-world lambda function. But if I import it from '/opt/utils' this works just fine.

Only after running `sam build` and running `sam local start-api` targetting '.aws-sam/build/template.yaml does '/opt/nodejs' work. But this implies compromising the ability to edit my code without having to re-build the application each time.

I found a workarround for this by extending the NODE_PATH variable to include /opt and /opt/nodejs paths, so now I can just import my layer from 'utils'. But I believe this should be handled directly by the cli, so that we don't need to be overriding that variable.

I hope this informations helps clarifying my issue.

**Comment by dillonsvincent** [comment]
I'm also having an issue with accessing Layers when running the api locally
`sam local start-api`

I get a runtime error in my lambda function when I debug
`{"errorType":"Runtime.ImportModuleError","errorMessage":"Error: Cannot find module '/opt/nodejs/testlayer.js`

When running my nodejs lambda function using invoke I'm able to require my layer using the same method used when running on AWS
`sam local invoke TestFunctionWithLayer --event events/event.json`

Code Inside my Function to import layer
`var Layer = require("/opt/nodejs/testlayer.js");`

This above line fails on `sam local start-api` but has success with `sam local invoke` and when running on AWS Servers afte deployed.

**Comment by mndeveci** [comment]
@dillonsvincent & @santiperone

When SAM CLI builds a layer, it puts layer contents into correct folder which is defined [here](https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html#configuration-layers-path). If you try to run `sam local -t template.yaml` which points to the template that is not built, you will have this problem because your source files are not in the correct folder.

What I can recommend is to use `--parallel` and `--cached` flags for `sam build` command, which should improve the performance of build operation, and should give you faster feedback loop.

You can also try to use SAM Accelerate features like running `sam sync --watch --stack-name {stack-name}`, which will deploy your application into a stack, and will allow you to quickly update your function code and test it in the cloud.

Please let us know if these won't resolve your issue.

**Comment by jfuss** [comment]
Closing as this looks to be resolved. Please re-open or open a new issue and link to this one if follow up is needed, as we don't tend to monitor closed issues.
